ErikBagratyan/File_Problem

```diff
@@ -0,0 +1,48 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <algorithm>
+#include <map>
+#include <vector>
+
+using namespace std;
+
+bool myFunction(const pair<string, int>& num1, const pair<string, int>& num2)
+{
+	return (num1.second > num2.second);
+}
+
+int main()
+{
+	map<string, int> mp;
+	fstream fin;
+	fin.open("war+peace.txt", fstream::in | fstream::out | fstream::app);
+	if (!fin.is_open())
+	{
+		cout << "Error OPEN File\n";
+	}
+	else {
+		char ch;
+		string str = "";
+		while (!fin.eof())
+		{
+			str = "";
+			while (fin.get(ch) && ch != ' ' && ch != '\n') str += ch;
+			if(str != "") mp[str]++;
+		}
+	}
+	fin.close();
+
+	vector<pair<string, int>> vec;
+	for (auto it : mp)
+	{
+		vec.push_back(it);
+	}
+	sort(vec.begin(), vec.end(), myFunction);
```

Comment:
it is useful to know that in such cases whereby you only need a function to pass to an algorithm or any similar cases you can use lambda functions. They are basically unnamed functions that you can define in place. That frees you from the burden of writing a function, name it, etc.

`std::sort(vec.begin(), vec.end(), [](const auto &x1, const auto &x2)
              { return x1.second > x2.second; });`